design/ceph-mon-pv.md

```diff
@@ -0,0 +1,139 @@
+}
+```
+
+Unlike the storage requirements of OSDs (xf: StorageClassDeviceSets) it is
+reasonable to use default requirements for monitors (e.g. 5-10 GB). This means
+that a minimal claim may be specified using only the requested storage class in
+which case default resource requests will be used.
+
+```yaml
+volumeClaimTemplates:
+- spec:
+    storageClassName: gp2-ebs
+...
+volumeClaimTemplates:
+- spec:
+    storageClassName: local-storage
+```
+
+It is also reasonable for monitors to use storage in the form of a directory on
+the host's root file system. This behavior effectively provides something like
+Rook-specific dynamic HostPath provisioning and is the current default behavior
+in Rook. In order to continue to support this level of operational simplicity,
+the volume claim template itself can be left unspecified entirely to request the
+existing default behavior. Note that the storage class `local-storage` is not
+necessarily equivalent---there are a variety of implementations with the most
+popular requiring static provisioning [0, 1].
+
+[0]: https://github.com/rancher/local-path-provisioner
+[1]: https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner
+
+## Non-PV monitor storage
+
+Not all monitor storage requirements should be satisfied using PV storage. Most
+notably log and crash data. These data may continue to be stored in HostPath
+volume mounts. This will continue to be based on `dataDirHostPath`.
+
+## Host networking
+
+Currently Ceph monitor pods may be deployed using host-based networking, or
+placed behind a service using a portable network address. This poses a challenge
+when migrating a monitor (and it's PV) to a new node which has been deployed
+using host-based networking as the monitor's address will no longer valid.
+
+Support for migration of monitors with host-based networking is future work. To
+simplify the initial implementation, host-based networking will not be supported
+for non-local PVs.
+
+# StatefulSet pod management
+
+Monitor pods will be managed by a StatefulSet rather than Deployments which are
+currently use. This change offers (1) simplified naming for DNS and pods (e.g.
+`ceph-mon-a` rather than `ceph-mon-a-3ab423fb`), and (2) StatefulSets are able
+to handle much of the complexity of managing PVs including the mechanics of
+PV/pod association and scheduling.
+
+StatefulSets will be used to manage monitors pods in a 1-1 mapping between a pod
+and StatefulSet. This choice is made to retain maximum flexibility on naming,
+placement, and life cycle management of the pods by Rook.
+
+## Scheduling
+
+Rook currently makes explicit scheduling decisions for monitors. Standard node
+selector is used to control placement for non-local PVs (or dynamically
+provisions local PVs [0]). For statically provisioned local PVs, the Rook
+monitor scheduler will likely need to be made aware of available PVs.
+
+If we want to continue to support the current behavior of Rook behavior
```

Comment:
Is it me or this sentence is strange:

> support the current behavior of Rook behavior effectively like a

Reply:
Probably should be "the current behavior of Rook"?

Reply:
I'm also confused here as to what "current behavior" we are talking about. I think some more context explanation would be good here.

Reply:
Sorry for the confusion... the _current behavior of rook_ I was referring to is that monitors are always deployed on a directory provisioned on the host's file system. This is effectively dynamic provisioning. There are two ways to achieve this mode of operation (important for clusters lacking non-local PV support)--continue using HostPath or dynamically provision a local PV backed by a local directory.